python/ray/serve/scripts.py

```diff
@@ -286,68 +289,191 @@ def run(
+parser = argparse.ArgumentParser()
+parser.add_argument('--config-or-import-path')
+parser.add_argument('--host')
+parser.add_argument('--port')
+parser.add_argument('--blocking', action='store_true')
+parser.add_argument('--gradio', action='store_true')
+args = parser.parse_args()
+
+if pathlib.Path(args.config_or_import_path).is_file():
+    config_path = args.config_or_import_path
+    cli_logger.print(f'Deploying from config file: "{config_path}".')
+
+    with open(config_path, 'r') as config_file:
+        config = ServeApplicationSchema.parse_obj(yaml.safe_load(config_file))
+    is_config = True
+else:
+    import_path = args.config_or_import_path
+    cli_logger.print(f'Deploying from import path: "{import_path}".')
+    node = import_attr(import_path)
+    is_config = False
+
+if is_config:
+    client = _private_api.serve_start(
+        detached=True,
+        http_options={
+            'host': config.host,
+            'port': config.port,
+            'location': 'EveryNode',
+        },
+    )
+else:
+    client = _private_api.serve_start(
+        detached=True,
+        http_options={'host': args.host, 'port': args.port, 'location': 'EveryNode'},
+    )
 
+try:
     if is_config:
-        client = _private_api.serve_start(
-            detached=True,
-            http_options={
-                "host": config.host,
-                "port": config.port,
-                "location": "EveryNode",
-            },
-        )
+        client.deploy_app(config, _blocking=args.gradio)
+        cli_logger.success('Submitted deploy config successfully.')
+        if args.gradio:
+            handle = serve.get_deployment('DAGDriver').get_handle()
     else:
-        client = _private_api.serve_start(
-            detached=True,
-            http_options={"host": host, "port": port, "location": "EveryNode"},
+        handle = serve.run(node, host=args.host, port=args.port)
+        cli_logger.success('Deployed Serve app successfully.')
+
+    if args.gradio:
+        from ray.serve.experimental.gradio_visualize_graph import GraphVisualizer
+
+        visualizer = GraphVisualizer()
+        visualizer.visualize_with_gradio(handle)
+    else:
+        if args.blocking:
+            while True:
+                # Block, letting Ray print logs to the terminal.
+                time.sleep(10)
+
+except KeyboardInterrupt:
+    serve.shutdown()
+    sys.exit()
+"""
+        final_runtime_env = parse_runtime_env_args(
+            runtime_env=runtime_env,
+            runtime_env_json=runtime_env_json,
+            working_dir=working_dir,
+        )
+        if "env_vars" not in final_runtime_env:
+            final_runtime_env["env_vars"] = {}
+        final_runtime_env["env_vars"]["RAY_JOB_STOP_SIGNAL"] = "SIGINT"
+
+        client = JobSubmissionClient(address)
+        submission_id = client.submit_job(
+            entrypoint=(
+                f'python -c "{script}" '
```

Comment:
Actually random thought, could we even just forward the `serve run` command directly to JobSubmission and just reuse the existing code?
Like `entrypoint = 'serve run --config-or-import-path=...'` which would then run on a cluster node.

Or is that more complicated?

Reply:
That's a good idea, I'll try it!